src/app/admin/components/courthouses/create-courthouse/create-update-courthouse-form/create-update-courthouse-form.component.ts

```diff
@@ -0,0 +1,162 @@
+import { CourthouseService } from '@services/courthouses/courthouses.service';
+import { FormService } from '@services/form/form.service';
+import {
+  courthouseNameExistsValidator,
+  displayNameExistsValidator,
+  valueIsUndefined,
+} from '@validators/courthouse.validator';
+
+const controlErrors: FieldErrors = {
+  courthouseName: {
+    required: 'Enter a courthouse code',
+    courthouseNameExists: 'The courthouse code you entered exists already',
+  },
+  displayName: {
+    required: 'Enter a display name',
+    displayNameExists: 'The display name you entered exists already',
+  },
+  regionId: {
+    required: 'Select a region',
+  },
+};
+
+@Component({
+  selector: 'app-create-update-courthouse-form',
+  standalone: true,
+  imports: [ReactiveFormsModule, DataTableComponent],
+  templateUrl: './create-update-courthouse-form.component.html',
+  styleUrl: './create-update-courthouse-form.component.scss',
+})
+export class CreateUpdateCourthouseFormComponent implements OnInit {
+  @Output() submitForm = new EventEmitter<CreateUpdateCourthouseFormValues>();
+  @Output() cancel = new EventEmitter<void>();
+  @Output() errors = new EventEmitter<ErrorSummaryEntry[]>();
+
+  @Input() updateCourthouse: CreateUpdateCourthouseFormValues | null = null;
+  @Input() regions!: Region[];
+  @Input() companies!: SecurityGroup[];
+
+  selectedCompanies: SecurityGroup[] = [];
+  selectedCompany: SecurityGroup | undefined = undefined;
+
+  nameExistsValidator = courthouseNameExistsValidator();
+  displayNameExistsValidator = displayNameExistsValidator();
+  valueIsUndefined = valueIsUndefined();
+  courthouseService = inject(CourthouseService);
+
+  fb = inject(FormBuilder);
+  formService = inject(FormService);
+  destroyRef = inject(DestroyRef);
+
+  formDefaultValues: CreateUpdateCourthouseFormValues = {
+    courthouseName: null,
+    displayName: null,
+    regionId: null,
+    securityGroupIds: [],
+  };
+
+  form = this.fb.group({
+    courthouseName: [this.formDefaultValues.courthouseName, [Validators.required], [this.nameExistsValidator]],
+    displayName: [this.formDefaultValues.displayName, [Validators.required], [this.displayNameExistsValidator]],
+    regionId: [this.formDefaultValues.regionId, [this.valueIsUndefined]],
+    securityGroupIds: [this.formDefaultValues.securityGroupIds],
+  });
+
+  ngOnInit(): void {
+    if (this.updateCourthouse) {
+      this.form.setValue({
+        courthouseName: this.updateCourthouse.courthouseName,
+        displayName: this.updateCourthouse.displayName,
+        regionId: this.updateCourthouse?.regionId?.toString() || '',
+        securityGroupIds: this.updateCourthouse?.securityGroupIds,
+      });
+
+      this.form.updateValueAndValidity();
+    }
+  }
+
+  onSubmit() {
+    this.form.markAllAsTouched();
+
+    // wait for async validation to complete
+    if (this.form.status === 'PENDING') {
+      return;
+    }
+
+    if (this.form.invalid) {
```

Comment:
`if (this.form.status === 'PENDING') {
      return;
    }`

Can be removed.

Use: 

`if (!this.form.valid)` instead of `if (this.form.invalid)`

This prevents the the user from clicking submit before the async validator resolves